# Problema 1 sección 4.8 ejercicio 6


In [2]:
import math
import sympy as sp

Suppose we collect data for a group of students statistics in a class with variables $X_1 = \text{hours}$ studied, $X_2 = \text{undergrad GPA}$, and Y = receive an A. We fit a logistic regression and produce estimated coefficient, $\hat{\beta_0}$ = −6, $\hat{\beta_1} = 0.05$, $\hat{\beta_2} = 1$.

(a) Estimate the probability that a student who studies for 40 h and has an undergrad GPA of 3.5 gets an A in the class.


In [7]:
b0 = -6
b1 = 0.05
b2 = 1
x1 = 40
x2 = 3.5
xb = (b0 + x1 * b1 + x2 * b2) * -1
e = math.exp(xb)

y = 1 / (1 + e)
y

0.3775406687981454

Podemos concluir que la probabilidad que el estudiante que estudió 40 horas teniendo un GPA antes de graduarse de 3.5 obtenga una A es del 37.754%


(b) How many hours would the student in part (a) need to study to have a 50% chance of getting an A in the class?


In [9]:
x1 = sp.symbols("x1")
y = 1 / (1 + sp.exp(-6 + 0.05 * x1 + 3.5))

eq = sp.Eq(y, 0.5)

solution = sp.solve(eq, x1)

print("El valor para x1 donde y = 0.5 es:", solution)

El valor para x1 donde y = 0.5 es: [50.0000000000000]


Prueba del resultado


In [10]:
b0 = -6
b1 = 0.05
b2 = 1
x1 = 50
x2 = 3.5
xb = (b0 + x1 * b1 + x2 * b2) * -1
e = math.exp(xb)

y = 1 / (1 + e)
y

0.5